Makefile

```diff
@@ -277,8 +277,10 @@ volume-plugin-test-img:
 test/goecho/goecho: .gopathok $(wildcard test/goecho/*.go)
 	$(GOCMD) build $(BUILDFLAGS) $(GO_LDFLAGS) '$(LDFLAGS_PODMAN)' -o $@ ./test/goecho
 
-test/version/version: .gopathok version/version.go
-	$(GO) build -o $@ ./test/version/
+# The ./test/version/version binary is executed in other make steps
+# so we have to make sure the version binary is built for NATIVE_GOARCH.
+test/version/version: version/version.go
+	GOARCH=$(NATIVE_GOARCH) $(GO) build -o $@ ./test/version/
```

Comment:
Is this on `main`?  If not, it probably should be.

Reply:
yes it is, i think i stole it from main during the cherrypick.